cds_ils/ldap/api.py

```diff
@@ -375,6 +364,68 @@ def sync_users():
     return (
         len(ldap_users),
         invenio_users_updated_count,
-        invenio_users_deleted_count,
         invenio_users_added_count,
     )
+
+
+def delete_users(dry_run=True):
+    """Delete users that are still in the DB but not in LDAP."""
+    raise NotImplementedError("not yet tested properly")
+    invenio_users_deleted_count = 0
+
+    # get all CERN users from LDAP
+    ldap_client = LdapClient()
+    ldap_users = ldap_client.get_primary_accounts()
+
+    if not ldap_users:
+        return 0, 0
+
+    # create a map by employeeID for fast lookup
+    ldap_users_map = {}
+    for ldap_user in ldap_users:
+        ldap_person_id = ldap_user_get(ldap_user, "employeeID")
+        ldap_users_map[ldap_person_id] = ldap_user
+
+    remote_accounts = RemoteAccount.query.all()
+
+    # get all Invenio remote accounts and prepare a list with needed info
+    invenio_users = []
+    for remote_account in remote_accounts:
+        invenio_users.append(
+            dict(
+                remote_account_id=remote_account.id,
+                remote_account_person_id=remote_account.extra_data[
+                    "person_id"
+                ],
```

Comment:
For consistency
```suggestion
                remote_account_person_id=remote_account.extra_data.get(
                    "person_id"
                ),
```

Reply:
No, this is on purpose. It should badly fail if not there.